[ENH] BEP018: Genetic information (#287)
bids-standard/bids-specification: src/04-modality-specific-files/08-genetic-descriptor.md

# Genetic Descriptor

> The extension was primarily developped by Cyril Pernet and Clara Moreau with contributions from Tom Nichols and Jessica Turner.

Reviewer: Do you have a "preprint" or blogpost or some other document where you describe the process a bit more in detail? It'd be cool to link it here instead of the list of names, because that would be more informative and provide more background ... akin to the publications in EEG, MEG, iEEG

Author: we gonna try a GigaScience editorial - but will take a little while as we also want to push a dataset

> The goal of the genetic descriptor is to link imaging and genetic data by providing minimal information about i) where to find genetic information associated with the imaging data and ii) what type of genetic information is available. This is necessary as genetic data are typically stored in dedicated repositories, separately from the imaging data.

Reviewer: I suggest to revise this paragraph to use a markdown list instead of the in-text i) ii) notation

Author: The goal of the genetic descriptor is to link imaging and genetic data. This is necessary as genetic data are typically stored in dedicated repositories, separately from the imaging data. The descriptor provides basics information about:  
* where to find genetic information associated with the imaging data  
* what type of genetic information is available

## dataset_description.json

> Two additional keys related to the genetic data can be added. The Key `GeneticDataBase` (MANDATORY) links to the name of the database and web address. The key `GeneticDescriptor` (OPTIONAL) refers to the descriptor (e.g. journal article) of the genetic data.

Reviewer: I suggest to revise this paragraph to use our words MUST(=mandatory), SHOULD(=recommended), and MAY(=optional), see [rfc2119](https://www.ietf.org/rfc/rfc2119.txt).

This would also solve the issue that the first sentence is a bit misleading, because it says "two keys **can** be added", which sounds like both are optional

Reviewer 2: Maybe `GeneticDataBase` could be mandatory if 'GeneticID' is present in any tsv files or if a genetic_info.json file is present.

Author: The Key `GeneticDataBase` must be added to link to the name of the database and web address. The key `GeneticDescriptor` may also be present refering to the descriptor (e.g. journal article) of the genetic data.

Author: 'GeneticID' is not mandatory if the imaging repo and genetic repo use the same ID (which the validator cannot check)

`dataset_description.json` example:

{  
  "Name": "xxxxx",  
  "BIDSVersion":  "xxxxx",  
  "License": "CC0",  
  "Authors": ["1st author", "2nd author"],  
  "Funding": "list your funding sources",  
  "GeneticDataBase": ["Name", "www",],  
  "GeneticDescriptor": ["data paper", "10.0.2.3/dfjj.10"]  

> }

Reviewer: For this example, you can use inline code formatting of Markdown. Use Three backticks to start a code paragraph ` ... then type which kind of code is following (JSON) ... then your example ... finally, three closing backticks.

```JSON
`` `JSON
{
"My": "example"
}
`` `
```

Please note that I intentionally left a blank between the backticks to prevent it from rendering and so that you can better see what I mean. See also the GitHub help: https://help.github.com/en/articles/creating-and-highlighting-code-blocks

---

Finally, once the formatting is done, please fill in the "xxx" fields with actual examples, a proper name, version, and database ... of course these don't have to make sense completely, but it'd be a more obvious example.

Author: `dataset_description.json` example:  
```JSON   
{  
  "Name": "Human Connectom Project",
  "BIDSVersion":  "1.2.0",
  "License": "CC0",
  "Authors": ["1st author", "2nd author"],
  "Funding": "list your funding sources", 
  "GeneticDataBase": ["Name", "dbGaP Study Accession: phs001364.v1.p1",],
  "GeneticDataBase": ["dbGaP", "phs001364.v1.p1"],
  "GeneticDescriptor": ["The HCP SNP dataset, FancyJOurnal"]
}  
```

> ## sub- and participant.tsv

Reviewer 3: Perhaps considering to change the header to follow some of our current formating. [an example location of my thoughts](https://bids-specification.readthedocs.io/en/latest/03-modality-agnostic-files.html#participants-file)

```suggestion
## Subject naming and Participants file
```

> One issue with having two repositories (one for imaging, one for genetics) is naming of subjects. A simple solution is to use the same ID in both datasets. Alternatively, one can encode the name used in the genetic database in the participant.tsv file. The column `GeneticID` (OPTIONAL) encodes the ID used for the same subject in the `GeneticDataBase` referred to in the dataset_description.json file.

Reviewer: ```suggestion
One issue with having two repositories (one for imaging, one for genetics) is naming of subjects. A simple solution is to use the same ID in both datasets. Alternatively, one can encode the name used in the genetic database in the participant.tsv file. The column `GeneticID` MAY encode the ID used for the same subject in the `GeneticDataBase` referred to in the dataset_description.json file.
```

Information about the presence/absence of specific genetic markers can be and should be shared simply in the participants.tsv file adding optopnal columns.

`participants.tsv` example:

```Text
participant_id  age sex group GeneticID	IDH Mutation
sub-control01 34  M control 124587	yes
sub-control02 12  F control 548936	yes
sub-patient01 33  F patient 489634	no
```

## genetic_info.json

> This file is the descriptor of the genetic information available either in the participant tsv file and/or the genetic database described in the dataset_description.json. The 'GeneticLevel' and 'SampleOrigin' are the only two mandatory fields.

Reviewer: ```suggestion
This file is the descriptor of the genetic information available either in the participant tsv file and/or the genetic database described in the dataset_description.json. The `GeneticLevel` and `SampleOrigin` are the only two mandatory fields.
```

Use backticks to format field names

| Field name   | Definition | Values |
| :----------- | :--------- | :------|
| GeneticLevel | MANDATORY Describes the level of analysis | `Genetic`, `Genomic`, `Epigenomic`, `Transcriptomic`, `Metabolomic`, or `Proteomic` |

> | AnalyticalApproach | OPTIONAL Methodology used to analyse the GeneticLevel | Value must be taken from [gapsolr](https://www.ncbi.nlm.nih.gov/projects/gapsolr/facets.html) under /Study/Molecular Data Type, for instance `SNP Genotypes (Array)` or `Methylation (CpG)` |

Reviewer: When you give an example of a filepath in text, please format that path using backticks like: `/this/is/a/path`

Author: don't got that? i used eg /Study/Molecular what's wrong

| SampleOrigin | MANDATORY Describes from which tissue the genetic information was extracted from | `blood`, `saliva`, `brain`, `csf`, `breast milk`, `bile`, `amniotic fluid`, `other biospecimen` |
| TissueOrigin | OPTIONAL Describes the type of tissue analyzed for SampleOrigin `brain` | `gray matter`, `white matter`, `csf`, `meninges`, `macrovascular` or `microvascular` |
| BrainLocation | OPTIONAL Refers to the location in space of the TissueOrigin | `MNI coordinate` or a `label` taken from the [Allen Brain Atlas](http://atlas.brain-map.org/atlas?atlas=265297125#atlas=265297125&plate=112360888&structure=4392&x=40348.15104166667&y=46928.75&zoom=-7&resolution=206.60&z=3) possibly `layer` to refer to layer-specific gene expression, which can also tie up with laminar fMRI |
| CellType | OPTIONAL Describes the type of cell analyzed | Value should come from the [cell ontology](http://obofoundry.org/ontology/cl.html) |

`genetic_info.json` example:

{  
  "GeneticLevel": "Genetic",  
  "AnalyticalApproach": "SNP Genotypes",  
  "SampleOrigin": "brain",  
  "TissueOrigin": "gray matter",  
  "CellType":  "neuron",  
  "BrainLocation": "[-30 -15 10]"  

> }

Reviewer: again, please format this as a JSON codeblock, as described in my comment above.

Author: `genetic_info.json` example:
```JSON   
{  
  "GeneticLevel": "Genetic",
  "AnalyticalApproach": "SNP Genotypes", "SampleOrigin": "brain",
  "TissueOrigin": "gray matter",
  "CellType":  "neuron",
  "BrainLocation": "[-30 -15 10]"
}
```

# Genetic Descriptor

The extension was primarily developped by Cyril Pernet and Clara Moreau with contributions from Tom Nichols and Jessica Turner.

> The goal of the genetic descriptor is to link imaging and genetic data. This is necessary as genetic data are typically stored in dedicated repositories, separately from the imaging data. The descriptor provides basics information about:

Author: done

# Genetic Descriptor

The extension was primarily developped by Cyril Pernet and Clara Moreau with contributions from Tom Nichols and Jessica Turner.

The goal of the genetic descriptor is to link imaging and genetic data. This is necessary as genetic data are typically stored in dedicated repositories, separately from the imaging data. The descriptor provides basics information about:
-   where to find genetic information associated with the imaging data
-   what type of genetic information is available

> 

Reviewer: ```suggestion
```

Author: comitted, thx

## dataset_description.json

The Key `GeneticDataBase` MUST be added to link to the name of the database and web address. The key `GeneticDescriptor` MAY also be present refering to the descriptor (e.g. journal article) of the genetic data.

`dataset_description.json` example:

> 

Reviewer: ```suggestion
```

| BrainLocation      | OPTIONAL Refers to the location in space of the TissueOrigin                     | `MNI coordinate` or a `label` taken from the [Allen Brain Atlas](http://atlas.brain-map.org/atlas?atlas=265297125#atlas=265297125&plate=112360888&structure=4392&x=40348.15104166667&y=46928.75&zoom=-7&resolution=206.60&z=3) possibly `layer` to refer to layer-specific gene expression, which can also tie up with laminar fMRI |
| CellType           | OPTIONAL Describes the type of cell analyzed                                     | Value should come from the [cell ontology](http://obofoundry.org/ontology/cl.html)                                                                                                                                                                                                                                                  |

`genetic_info.json` example:

> 

Reviewer: ```suggestion
```

# Genetic Descriptor

Support for genetic descriptors was developed as a
[BIDS Extension Proposal](https://github.com/bids-standard/bids-specification/blob/master/src/06-extensions.md#bids-extension-proposals).
The extension was primarily developped by Cyril Pernet and Clara Moreau
with contributions from Tom Nichols and Jessica Turner.

The goal of the genetic descriptor is to link imaging and genetic data. This is necessary as genetic data are typically stored in dedicated repositories, separately from the imaging data. The descriptor provides basics information about:

> -   where to find genetic information associated with the imaging data

Reviewer 4: No directory structure is specified for genetics data in this PR? Top-level [genetics] seems most intuitive.

Reviewer 5: indeed, as explained the goal is to indicate where to find genetic information associated with the imaging data and what type of genetic information is available

Author: indeed, as explained the goal is to indicate where to find genetic information associated with the imaging data and what type of genetic information is available -- so really not much but some metadata

Reviewer 4: Sorry, I'm relatively new to this, but isn't that a problem? Knowing where to get access to the data (dbGaP, phs001364.v1.p1) is useful for people, but less so for tools. If I'm using code to perform some analysis, the only data available to me through BIDS would be in the participants.tsv file (e.g. "IDH mutation" from line 59), which would need to be regenerated for different variants. In cases with large datasets and users with read-only access, that seems like a serious limitation.
Given that the participants.tsv + genetic_info.json files give a lot of information that is both user- and machine-readable (and often not as readily-available as proposed here), it seems to me that adding a directory structure spec would be a huge improvement.

If the intent is to provide user-readable metadata, then I agree that specifying a directory structure is beyond the scope of this BEP (in which case you can mark this as resolved).

Author: well yes it is - but there no way this can happen ; genetic data are protected by ethic governance because they are personal data and this cannot be public the way we can for imaging == >> can't do what you are asking for in 99% of cases - so yeh we'll stick to user-readable metadata

Reviewer 4: Isn't sharing the data a separate issue? The data doesn't need to be public for it to be BIDS-compliant. If somebody has collected imaging + genetics data, they'll have access to the data and the question of ethics has already been addressed. BIDS specifies how to structure imaging data regardless of whether the data is shared (Nifti format, directories for anat & func, etc.), but the genetics BEP doesn't specify how to structure genetics data (e.g. data format: text and/or binary PLINK, filenames for different chromosomes, imputed & derived data, etc.).
Since participants.tsv provides the imaging <-> genetics link, the standard implicitly assumes that the genetics data is available while not specifying how it is supposed to be formatted.

Reviewer 5: that's just not what we made it for -- now once merged nothing say you can work on version 2

-   what type of genetic information is available

## dataset_description.json

In order to link a genetic database entry, the key `GeneticDataBase` MUST be present and

> the value is an object with the following fields:

Reviewer 5: can you point out the inconsistencies please?

> the value is an object with the following fields:

Author: can you point out the inconsistencies please? would help resolving - thx
I'm working on the HCP data -- you could push a 0Kb data example from UK biobank (better ask them though)

> the value is an object with the following fields:

Reviewer 4: Current example JSON:
```JSON
{
  "Name": "Human Connectom Project",
  "BIDSVersion":  "1.2.0",
  "License": "CC0",
  "Authors": ["1st author", "2nd author"],
  "Funding": "list your funding sources",
  "GeneticDataBase": ["dbGaP", "phs001364.v1.p1"],
  "GeneticDescriptor": ["The HCP SNP dataset, FancyJournal"]
}
```
What it sounds the JSON should look like from the description ("The key GeneticDataBase is an object with the following fields: Genetics.Dataset [...]"):
```JSON
{
  "Name": "Human Connectome Project",
  "BIDSVersion":  "1.2.0",
  "License": "CC0",
  "Authors": ["1st author", "2nd author"],
  "Funding": "list your funding sources",
  "GeneticDataBase": {
    "Genetics": {
     "Dataset": "dataset",
     "Database": "database",
     "Descriptors": "descriptors"
     }
    }
}
```

Removing the "Genetics . " prefix in lines 19-21 would resolve it, I think.

> the value is an object with the following fields:

Reviewer 6: Ah, that proposal was actually to replace `GeneticDataBase` and `GeneticDescriptor` with a `Genetics` object containing `Dataset`, `Database` and `Descriptors`.

> the value is an object with the following fields:

Author: maybe - the reason descriptor was 'out' was to make it the same as we do the descriptor for the imaging part ; I''m ok with that though

> the value is an object with the following fields:

Author: updated as object
